packages/erk-shared/src/erk_shared/context/helpers.py

```diff
@@ -0,0 +1,320 @@
+"""Helper functions for accessing context dependencies with LBYL checks.
+
+This module provides getter functions that encapsulate the "Look Before You Leap"
+pattern for accessing dependencies from the ErkContext. These functions:
+
+1. Check that context is initialized
+2. Return the typed dependency
+3. Exit with clear error message if context is missing
+
+This eliminates code duplication across kit CLI commands.
+"""
+
+from pathlib import Path
+
+import click
+
+from erk_shared.context.types import NoRepoSentinel
+from erk_shared.extraction.claude_code_session_store import ClaudeCodeSessionStore
+from erk_shared.git.abc import Git
+from erk_shared.github.abc import GitHub
+from erk_shared.github.issues import GitHubIssues
+from erk_shared.project_discovery import discover_project
+from erk_shared.prompt_executor import PromptExecutor
+
+
+def require_issues(ctx: click.Context) -> GitHubIssues:
+    """Get GitHub Issues from context, exiting with error if not initialized.
+
+    Uses LBYL pattern to check context before accessing. If context is not
+    initialized (ctx.obj is None), prints error to stderr and exits with code 1.
+
+    Handles both ErkContext (with .issues) and simple test contexts (with .github_issues).
+
+    Args:
+        ctx: Click context (must have ErkContext or compatible context in ctx.obj)
+
+    Returns:
+        GitHubIssues instance from context
+
+    Raises:
+        SystemExit: If context not initialized (exits with code 1)
+
+    Example:
+        >>> @click.command()
+        >>> @click.pass_context
+        >>> def my_command(ctx: click.Context) -> None:
+        ...     issues = require_issues(ctx)
+        ...     issues.add_comment(repo_root, issue_number, body)
+    """
+    if ctx.obj is None:
+        click.echo("Error: Context not initialized", err=True)
+        raise SystemExit(1)
+
+    # Handle simple test context with github_issues
+    if hasattr(ctx.obj, "github_issues"):
+        return ctx.obj.github_issues
+
+    # Handle ErkContext with issues
+    if hasattr(ctx.obj, "issues"):
+        return ctx.obj.issues
+
+    click.echo("Error: Context missing issues", err=True)
+    raise SystemExit(1)
```

Comment:
\# Handle simple test context with github_issues

    if hasattr(ctx.obj, "github_issues"):

return ctx.obj.github_issues  
  
we should NOT require this anymore  
\----  
    # Handle ErkContext with issues

    if hasattr(ctx.obj, "issues"):



we should test ctx.obj to see if it is is an ErkContext and then type narrow appropriately